tests/integrations/test_dsm.py

```diff
@@ -370,3 +533,84 @@ def assert_checkpoint_presence(hash_, parent_hash, tags):
 
         logger.error("Checkpoint not found 🚨")
         raise ValueError("Checkpoint has not been found, please have a look in logs")
+
+    @staticmethod
+    def produce_rabbitmq_message_v1_propagation(queue, exchange):
+        # Create a RabbitMQ client
+        conn = kombu.Connection("amqp://127.0.0.1:5672")
+        conn.connect()
+        producer = conn.Producer()
+
+        task_queue = kombu.Queue(queue, kombu.Exchange(exchange), routing_key=queue)
+
+        headers = {
+            "dd-pathway-ctx": b"\xd7I\xd5\xcdy\x9e*\x80\xe6\x86\x8a\xa6\xb6c\xe6\x86\x8a\xa6\xb6c"  # encoded V1 pathway from dd-trace-py, pathway hash is: 9235368231858162135
+        }
+        to_publish = {"message": "DSM Pathway Encoding V1 Test"}
+
+        try:
+            producer.publish(
+                to_publish,
+                exchange=task_queue.exchange,
+                routing_key=task_queue.routing_key,
+                declare=[task_queue],
+                headers=headers,
+            )
+            logging.info("System Tests RabbitMQ message using V1 DSM Pathway Encoding sent successfully")
+            return "ok"
+        except Exception as e:
+            logging.info(f"Error during DSM RabbitMQ publish message using V1 DSM Pathway Encoding: {e}")
+            return "error"
+
+    @staticmethod
+    def produce_rabbitmq_message_v2_base64_propagation(queue, exchange):
+        # Create a RabbitMQ client
+        conn = kombu.Connection("amqp://127.0.0.1:5672")
```

Comment:
I guess in this case it has no impact, but you should close the connection.

Reply:
updated to close all rabbitmq connections